src/node.cc

```diff
@@ -2261,6 +2261,56 @@ static void Uptime(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+static void GetResourceUsage(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  uv_rusage_t rusage;
+  int err = uv_getrusage(&rusage);
+  if (err)
+    return env->ThrowUVException(err, "uv_getrusage");
+
+  Local<Object> resource_usage = Object::New(env->isolate());
```

Comment:
@mscdex If you're confident in the benchmark results, feel free to push the most performant implementation directly to this branch if you want. (Or open a separate PR and close this one. Either way!)